### PR TITLE
Correspondence Reference: Limits

### LINKED ISSUE #3027: Update Limits.md documentation
This documentation is very outdated and should be updated for the latest limits. Additionally, I want to eventually add full details on how the limit is calculated on each underlying platform. In #1962 for example, the computation is fairly complex.

### PR DESCRIPTION
**Pull request by kainino0x**
**For rendered version, see preview link below**

Documents how to compute limits on each platform. Subsumes Limits.md.

Fixes #3027

### CODE REVIEW

**Comment by kainino0x** [comment]
@cwfitzgerald you may also be interested in this

**Comment by cwfitzgerald** [comment]
This is great to have!

Looks terrible on dark mode safari/chrome though 
image

**Comment by cwfitzgerald** [comment]
Good stuff. Do note that FF has had trouble with buffer sizes above 2GB: https://github.com/gfx-rs/wgpu/pull/2796

File: correspondence/index.bs [hunk 1]
```diff
@@ -46,4 +46,196 @@ Issue: Migrate from `APICorrespondence.md`.
...
+            <td>[Constants](https://docs.microsoft.com/en-us/windows/win32/direct3d12/constants),
+                [Hardware Tiers](https://docs.microsoft.com/en-us/windows/win32/direct3d12/hardware-support),
+                [Root Signature Limits](https://docs.microsoft.com/en-us/windows/win32/direct3d12/root-signature-limits)
+    </thead>
+    <tr>
+        <th class=issue>need a max buffer size limit
+        <td>[#1371](https://github.com/gpuweb/gpuweb/issues/1371)
+        <td>`maxMemoryAllocationSize`
+        <td>`MTLDevice.maxBufferLength`
+        <td class=issue>*No documented limit?*
```
Comment (toji) [inline correspondence/index.bs:76]: Nit: Noticed a strange style quirk while reviewing the preview. Any cell that is marked as an issue has a white left border, while none of the other cells do.

Screen Shot 2022-06-30 at 3 28 08 PM

Not a big deal, but it does look a little weird.

File: correspondence/index.bs [hunk 2]
```diff
@@ -46,4 +46,196 @@ Issue: Migrate from `APICorrespondence.md`.
...
+        <th>`maxSampledTexturesPerShaderStage`
+        <td>[#409](https://github.com/gpuweb/gpuweb/issues/409)
+        <td>`maxPerStageDescriptorSampledImages`
+        <td rowspan=2>*Strategy-dependent.* Allocate `Maximum number of entries in the texture argument table, per graphics or kernel function` across these two limits.
+        <td>`Maximum number of Shader Resource Views in all descriptor tables per shader stage`
+    <tr>
+        <th>`maxStorageTexturesPerShaderStage`
+        <td>[#409](https://github.com/gpuweb/gpuweb/issues/409)
+        <td>`maxPerStageDescriptorStorageImages`
+        <td rowspan=2>*Strategy-dependent.* Allocate `Maximum number of Unordered Access Views in all descriptor tables across all stages` across these two limits.
```
Comment (toji) [inline correspondence/index.bs:137]: This rowspan, when paired with two offset rowspans in the Metal column, is a little awkward to visually to parse in the preview. I don't really have a suggestion for resolving it, but felt like it was worth pointing out anyway.

Screen Shot 2022-06-30 at 3 24 38 PM

**Review by toji — APPROVED** [review]
LGTM, thanks for putting this together! Two minor style nits, though I don't consider either of them to be blocking.

File: correspondence/index.bs (hunk 1 above)
Comment (kainino0x) [reply correspondence/index.bs:76]: Seems the issue class is not really intended to work with td, it just happened to look pretty nice. I'll just turn these into normal issues.

File: correspondence/index.bs (hunk 2 above)
Comment (kainino0x) [reply correspondence/index.bs:137]: An alternative would be to do these as footnotes, no rowspans, put "[1]"/"[2]"/"[3]" in each cell and then put the text at the bottom.

**Pull request merged**